tools/query_breakdown/src/main/java/com/google/bigquery/InputReader.java

```diff
@@ -0,0 +1,43 @@
+package com.google.bigquery;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.List;
+
+/**
+ * This class will take care of the input handling logic, essentially parsing the input document
+ * into queries and data-cleaning if needed.
+ */
+public class InputReader {
+
+  /**
+   * This method will take in a txt file name, use BufferedReader to parse the input, and return
+   * all the queries in a string format
+   */
+  public static String readInput(String filename) throws IOException {
+    List<String> lines = Files.readAllLines(Paths.get(filename));
+
+    StringBuilder sb = new StringBuilder();
+    for (String line : lines) {
+      sb.append(line);
+    }
+
+    String parsedInput = sb.toString();
+    int len = parsedInput.length();
+    boolean lastSemicolon = false;
+
+    // deals with case where last query ends with a semicolon
+    if (parsedInput.charAt(len - 1) == ';') {
```

Comment:
Why not first replace and then substring? In this case you end up doing substring, replace and append.

Reply:
fixed accordingly!